include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -663,17 +674,17 @@ struct __result_and_scratch_storage : __result_and_scratch_storage_base
     // Note: this member function assumes the result is *ready*, since the __future has already
     // waited on the relevant event.
     _T
-    __get_value(size_t idx = 0) const
+    __get_value() const
```

Comment:
We really shouldn't support more then one value in our code?

Reply:
Theoretically, `future` type has one event and N other parameters.
But, practically is used only one parameter as value. Even there is a case with two values and method `__get_value(i)` may be useful, it introduces a performance penalty because "wait"  is calling on each `__get_value` call.
So, if we want to continue using future object, it should be re-designed in the future.

Reply:
You are right, sure, it will calls `wait` in every `__get_value`, but for the same 'sycl::event'. Correct?

Reply:
Correct.

Reply:
So this mean that if we pack two numbers as two results (with indexes 0, 1) into one `__result_and_scratch_storage` probably we wouldn't  have performance impact described by you.

Reply:
As mentioned in the comment above this function, it is assumed that wait has already been called.  So one could wait on an event, and then get multiple values from the result and scratch storage. 
I suppose this multi-element access is not currently used from a quick look through the repo, but the idea was because we are using a singular value type for `result_and_scratch_storage`, so that it can be efficiently packed together with temporary storage in a single container.  This means the result wouldnt be able to be a tuple or something similar without the temp value type also being the same.

This change seems orthogonal to the PR, and we may want to separate it to its own PR / issue.

Reply:
As mentioned below, I don't think we should alter / remove this functionality.  What it is being replaced with is not a long term solution, so lets wait to make changes here until we do a real examination and refactor in the next release.